modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/fds_view/Filters.tsx

```diff
@@ -0,0 +1,331 @@
+								)}
+							>
+								<ClayIcon symbol="question-circle-full" />
+							</span>
+						</label>
+
+						<ClayInput
+							aria-label={Liferay.Language.get('label')}
+							name={namespace + 'label'}
+							onChange={(event) => setLabel(event.target.value)}
+							placeholder={
+								fields.find(
+									(item) => item.name === selectedField
+								)?.label
+							}
+							value={label}
+						/>
+					</ClayForm.Group>
+				)}
+			</ClayModal.Body>
+
+			<ClayModal.Footer
+				last={
+					<ClayButton.Group spaced>
+						<ClayButton onClick={handleFilterSave}>
+							{Liferay.Language.get('save')}
+						</ClayButton>
+
+						<ClayButton
+							displayType="secondary"
+							onClick={() => closeModal()}
+						>
+							{Liferay.Language.get('cancel')}
+						</ClayButton>
+					</ClayButton.Group>
+				}
+			/>
+		</div>
+	);
+}
+
+interface IProps {
+	fdsView: FDSViewType;
+	fdsViewsURL: string;
+	namespace: string;
+}
+
+function Filters({fdsView, fdsViewsURL, namespace}: IProps) {
+	const [fields, setFields] = React.useState<Field[]>([]);
+	const [filters, setFilters] = React.useState<Filter[]>([]);
+	const [newFiltersOrder, setNewFiltersOrder] = React.useState<string>('');
+
+	const updateFDSFiltersOrder = async () => {
+		const response = await fetch(
+			`${API_URL.FDS_VIEWS}/by-external-reference-code/${fdsView.externalReferenceCode}`,
+			{
+				body: JSON.stringify({
+					fdsFiltersOrder: newFiltersOrder,
+				}),
+				headers: {
+					'Accept': 'application/json',
+					'Content-Type': 'application/json',
+				},
+				method: 'PATCH',
+			}
+		);
+
+		if (!response.ok) {
+			alertFailed();
+
+			return null;
+		}
+
+		const responseJSON = await response.json();
+
+		const fdsFiltersOrder = responseJSON?.fdsFiltersOrder;
+
+		if (fdsFiltersOrder && fdsFiltersOrder === newFiltersOrder) {
+			alertSuccess();
+
+			setNewFiltersOrder('');
+		}
+		else {
+			alertFailed();
+		}
+	};
+
+	const onCreationButtonClick = () =>
+		openModal({
+			contentComponent: ({closeModal}: {closeModal: Function}) => (
+				<AddFDSFilterModalContent
+					closeModal={closeModal}
+					fdsView={fdsView}
+					fields={fields}
+					namespace={namespace}
+					onSave={(newfilter) => setFilters([...filters, newfilter])}
+				/>
+			),
+		});
+
+	React.useEffect(() => {
+		const getFilters = async () => {
+			const response = await fetch(
+				`${API_URL.FDS_FILTERS}?filter=(${OBJECT_RELATIONSHIP.FDS_VIEW_FDS_FILTER_ID} eq '${fdsView.id}')&nestedFields=${OBJECT_RELATIONSHIP.FDS_VIEW_FDS_FILTER}`
+			);
+
+			const responseJSON = await response.json();
+
+			let filtersOrderer = responseJSON.items as Filter[];
+
+			if (fdsView.fdsFiltersOrder) {
+				filtersOrderer = fdsView.fdsFiltersOrder
+					.split(',')
+					.map((fdsFilterId) =>
+						filtersOrderer.find(
+							(filter) => filter.id === Number(fdsFilterId)
+						)
+					)
+					.filter(Boolean) as Filter[];
```

Comment:
What dark sorcery is this? :thinking:

Reply:
:warning:  [Warning: You should rarely find yourself using Boolean as a constructor](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Boolean#boolean_coercion:~:text=Warning%3A%20You%20should%20rarely%20find%20yourself%20using%20Boolean%20as%20a%20constructor.)??